ci_scripts/generate_news.py

```diff
@@ -1,64 +1,86 @@
-    """Writes out newsfile if significant version bump"""
-    last_known = '0'
-    if os.path.isfile(metafile):
-        with open(metafile) as fh:
-            last_known = fh.read()
-
-    import mbed_cloud
-    current = mbed_cloud.__version__
-
-    # how significant a change in version scheme should trigger a new changelog entry
-    # (api major, api minor, sdk major, sdk minor, sdk patch)
-    sigfigs = 4
-    current_version = LooseVersion(current).version
-    last_known_version = LooseVersion(last_known).version
-
-    should_towncrier = current_version[:sigfigs] != last_known_version[:sigfigs]
-
-    print('%s -- %s :: current vs previous changelog build' % (current, last_known))
-    if should_towncrier:
-        print('%s >> %s :: running changelog build' % (current, last_known))
+
+def version_project(commit_type: CommitType):
+    """
+    Versions the project
+    :param commit_type: states what is the type of the commit
+    :type: CommitType
+    """
+
+    use_news_files = commit_type in [CommitType.BETA, CommitType.RELEASE]
+
+    new_version = _calculate_version(commit_type, use_news_files)
+    _generate_changelog(new_version, use_news_files)
+
+
+def _calculate_version(commit_type: CommitType, use_news_files: bool):
+    BUMP_TYPES = {CommitType.DEVELOPMENT: "build",
+                  CommitType.BETA: "prerelease"}
+    is_release = commit_type == CommitType.RELEASE
+    enable_file_triggers = True if use_news_files else None
+    bump = BUMP_TYPES.get(commit_type)
+    old, new_version, updates = auto_version_tool.main(
+        release=is_release,
+        enable_file_triggers=enable_file_triggers,
+        bump=bump,
+        config_path=PROJECT_CONFIG,
+    )
+    logger.info(f'Version: {new_version}')
+    return new_version
+
+
+def _generate_changelog(version: str, use_news_files: bool):
+    if use_news_files:
+        logger.info(':: Generating a new changelog')
         subprocess.check_call(
-            ['towncrier', '--yes'],
-            cwd=os.path.join(PROJECT_ROOT, 'docs', 'changelog')
+            ['towncrier', '--yes', '--name=""', f'--version="{version}"']
         )
-        with open(metafile, 'w') as fh:
-            fh.write(current)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Versions the project.')
+    parser.add_argument('-r', '--release',
+                        help='Update the version for an official release',
+                        action='store_true')
+    parser.add_argument('-b', '--beta',
+                        help='Update the version for a pre-release',
+                        action='store_true')
+    parser.add_argument("-v", "--verbose", action="count", default=0,
+                        help="Verbosity, by default errors are reported.")
+    args = parser.parse_args()
+    commit_type = CommitType.DEVELOPMENT
+    if args.release:
+        commit_type = CommitType.RELEASE
+    if args.beta:
+        commit_type = CommitType.BETA
```

Comment:
Would it be clearer to use a single argument with choices, something like:

`parser.add_argument('--release', default=CommitType.DEVELOPMENT, choices=[CommitType.RELEASE, CommitType.BETA, CommitType.DEVELOPMENT], help='...') 
`